src/main/java/org/opensearch/flowframework/transport/WorkflowRequest.java

```diff
@@ -146,6 +164,7 @@ public WorkflowRequest(StreamInput in) throws IOException {
         if (this.updateFields) {
             this.params = Collections.emptyMap();
         }
+        this.reprovision = in.readBoolean();
```

Comment:
could it be enough to just have a it as an optional boolean? I also think like we mentioned in a few PRs in the past the `WorkflowRequest` is getting pretty crowded vs having a reprovisionrequest for example.

Reply:
@amitgalitz optional booleans still have a boolean in front of them to define the optionalness.

But if you're upgrading a 2.15 cluster to 2.16 and have mixed nodes and run a workflow, you'll get an exception here.

We pass a params map and I've already altered the code (for the field replacement) to use that params map, which is otherwise unused when provisioning.

Reply:
@dbwiddis I have followed the same process you did for the update fields flag for the reprovision boolean. If you could take another look to see if this satisfies your concern.